marketplace/plugins/apyhub/lib/query_operations.ts

```diff
@@ -0,0 +1,699 @@
+        throw error;
+    }
+}
+
+
+interface JobResponse {
+    status_url: string;
+    job_id: string;
+}
+
+interface JobStatusResponse {
+    data: {
+        attributes: {
+            status: string;
+            result: any;
+        }
+    }
+}
+
+
+
+export function base64ToPdfFile(base64String: string, filename: string): string {
+    try {
+        const base64Data = base64String.includes(',')
+            ? base64String.split(',')[1]
+            : base64String;
+
+        const pdfBuffer = Buffer.from(base64Data, 'base64');
+        const filePath = path.join(__dirname, filename);
+        fs.writeFileSync(filePath, pdfBuffer);
+
+        return filePath;
+    } catch (error) {
+        console.error('Error creating PDF file:', error);
+        throw new Error('Failed to create PDF file from Base64 string');
+    }
+}
+
+async function submitResumeJob(options: QueryOptions, sourceOptions: SourceOptions): Promise<JobResponse> {
+
+    const formData = new FormData();
+    const pdfFilePath = base64ToPdfFile(options.resume, 'resume.pdf');
+    formData.append('file', fs.createReadStream(pdfFilePath));
+    formData.append('language', options.language);
+
+    try {
+        const response = await axios.post(
+            'https://api.apyhub.com/sharpapi/api/v1/hr/parse_resume',
+            formData,
+            {
+                headers: {
+                    'apy-token': sourceOptions.apiKey,
+                    ...formData.getHeaders(),
+                    'content-type': `multipart/form-data; boundary=${formData.getBoundary()}`
+                }
+            }
+        );
+        console.log('Job submitted successfully, response:', response.data);
+        return response.data;
+    } catch (error) {
+        console.error('Error submitting resume job:', error.response?.data || error);
+        throw error;
+    }
+}
+
+async function checkJobStatus(jobId: string, apiKey: string): Promise<JobStatusResponse> {
+    try {
+        console.log('Starting resume parsing process');
+        const response = await axios.get(
+            `https://api.apyhub.com/sharpapi/api/v1/hr/parse_resume/job/status/${jobId}`,
+            {
+                headers: {
+                    'apy-token': apiKey,
+                    'Content-Type': 'application/json'
+                }
+            }
+        );
+        return response.data;
+    } catch (error) {
+        console.error('Error checking job status:', error);
+        throw error;
+    }
+}
+
+export async function getParsedResume(queryOptions: QueryOptions, sourceOptions: SourceOptions): Promise<any> {
+    try {
+        // Submit job
+        console.log('Starting resume parsing process');
+        const jobResponse = await submitResumeJob(queryOptions, sourceOptions);
+        let retries = 0;
+
+        // Poll for results
+        while (retries < MAX_RETRIES) {
+            console.log(`Polling attempt ${retries + 1}/${MAX_RETRIES}`);
```

Comment:
Could you remove all the `console.logs` in this file? `console.errors` are fine.
The logs will get cluttered otherwise.

Reply:
Done